src/components/PriorityBadge/PriorityBadge.tsx

```diff
@@ -0,0 +1,245 @@
+import * as React from 'react';
+import {
+    MdClose,
+    MdKeyboardArrowDown,
+    MdKeyboardArrowUp,
+    MdKeyboardDoubleArrowUp,
+} from 'react-icons/md';
+import { bem } from '../../utils';
+import { Dropdown } from '../Dropdown';
+import { ListItem } from '../List';
+import { Text } from '../Text';
+import styles from './PriorityBadge.scss';
+
+export type Priority = 'mandatory' | 'important' | 'optional' | 'exclude';
+
+export interface Option {
+    value: string;
+    label: string;
+}
+
+export interface Props extends Omit<React.HTMLAttributes<HTMLDivElement>, 'onChange'> {
+    /**
+     * Children nodes to be rendered within the component,
+     * specifically used to display the selected value from parent component.
+     */
+    children: React.ReactNode;
+    /**
+     * Optional object containing labels for different priorities, used primarily for localization.
+     */
+    priorityLabels?: {
+        mandatory?: string;
+        important?: string;
+        optional?: string;
+        exclude?: string;
+    };
+    /**
+     * Current priority value that indicates the importance of the component.
+     * Uses the Priority type which can be 'mandatory', 'important', 'optional', or 'exclude'.
+     */
+    priority: Priority;
+    /**
+     * Optional object specifying the currently selected option with value and label properties.
+     */
+    option?: Option;
+    /**
+     * Array of options available for selection.
+     * Each option is an object with a value and a label.
+     */
+    optionList: Array<Option>;
+    /**
+     * Optional header title for the options list when displayed, such as in a dropdown.
+     */
+    optionListHeader?: string;
+    /**
+     * Function to be called when the close button is clicked.
+     */
+    onClose?: (e: React.KeyboardEvent | React.MouseEvent) => void;
+    /**
+     * Callback function triggered when a new priority is selected.
+     * Provides the newly selected priority as an argument.
+     */
+    onPriorityChange?: (newPriority: Priority) => void;
```

Comment:
I'm wondering why this function is optional? I believed `PriorityBadge` was actually designed for that purpose. Even component has that name.

Reply:
It's optional because the badge can be without priority button, Dima added it to the AC later. I was also thinking that the name doesn't make any sense in this case, maybe I should rename it to something like `SelectBadge`? 🤔 What do you think?